server/rate_counter.go

```diff
@@ -0,0 +1,38 @@
+package server
```

Comment:
Added copyright.

Utils mostly contains functions, where rate counter is "a class". (sry I'm an OOP guy :) 

Still move to utils ?

Reply:
If you keep this file, which is fine, change copyright date to simply 2021. We have used (maybe incorrectly?) the logic that the date range if based on the file creation/last modification. So since you add this file now, just put 2021.